examples/other/hirez_screenshot.py

```diff
@@ -0,0 +1,262 @@
+            marker, anchor="middle-right", font_size=20, screen_space=True
+        ),
+        gfx.TextMaterial("#000"),
+    )
+    text.local.y = -y
+    text.local.x = 0
+    scene.add(text)
+
+
+camera = gfx.OrthographicCamera()
+camera.show_object(scene, scale=0.8)
+
+camera_state = camera.get_state()
+
+# To help visualize the rendering ability, we draw two boxes
+# One entirely contained within the FOV of the main image, the other
+# entirely outside of the FOV of the main image.
+# By disabling AA, we should be able to use this to validate that we don't
+# render more than we should.
+line_thickness = 0.1
+box_width = camera_state["width"]
+box_height = camera_state["height"]
+logical_size = canvas.get_logical_size()
+if logical_size[0] > logical_size[1]:
+    box_width *= logical_size[0] / logical_size[1]
+else:
+    box_height *= logical_size[1] / logical_size[0]
+inner_lines_geometry = gfx.box_geometry(
+    width=box_width - line_thickness,
+    height=box_height - line_thickness,
+)
+
+
+inner_lines_geometry.positions.data[..., 0] += camera_state["position"][0]
+inner_lines_geometry.positions.data[..., 1] += camera_state["position"][1]
+
+outer_lines_geometry = gfx.Geometry(
+    positions=[
+        [0, 0, 0],
+        [box_width + line_thickness, 0, 0],
+        [box_width + line_thickness, box_height + line_thickness, 0],
+        [0, box_height + line_thickness, 0],
+        [0, 0, 0],
+    ]
+)
+outer_lines_geometry.positions.data[..., 0] -= (box_width + line_thickness) / 2
+outer_lines_geometry.positions.data[..., 1] -= (box_height + line_thickness) / 2
+
+outer_lines_geometry.positions.data[..., 0] += camera_state["position"][0]
+outer_lines_geometry.positions.data[..., 1] += camera_state["position"][1]
+scene.add(
+    gfx.Line(
+        inner_lines_geometry,
+        gfx.LineMaterial(
+            color="blue", thickness=line_thickness, thickness_space="world", aa=False
+        ),
+    )
+)
+scene.add(
+    gfx.Line(
+        outer_lines_geometry,
+        gfx.LineMaterial(
+            color="red", thickness=line_thickness, thickness_space="world", aa=False
+        ),
+    )
+)
+
+
+## Tiling
+
+
+@canvas.request_draw
+def animate():
+    renderer.render(scene, camera)
+
+
+# Create snapshot of tiles.
+# A possible improvement would be to write each tile-row once it is captured,
+# so we never need the full image as one contiguous array. This would enable
+# creating massive screenshots even on machines with little RAM.
+rows = []
+for iy in range(upscale_factor):
+    row = []
+    for ix in range(upscale_factor):
+        camera.set_view_offset(
+            full_size[0],
+            full_size[1],
+            ix * tile_size[0],
+            iy * tile_size[1],
+            tile_size[0],
+            tile_size[1],
+        )
+        im = np.asarray(canvas.draw())
+        # im = im[:,:,:3]  # rgba -> rgb
+        row.append(im)
+    rows.append(np.column_stack(row))
+
+# Safe full image
+full_im = np.vstack(rows)
```

Comment:
```suggestion
full_im = np.block(rows)
```

For very very large images (which is the subject for in this PR), your are incuring 2 copies of all the renders"

1. for the column stack
2. The second for the vstack

By using modern (last 6 years) call to `np.block` it will do a single copy.
```
a = np.full((5, 4), fill_value=0)
b = np.full((5, 4), fill_value=1)
c = np.full((2, 4), fill_value=2)
d = np.full((2, 4), fill_value=3)

np.block([[a, b], [c, d]])
```

outputs
```
array([[0, 0, 0, 0, 1, 1, 1, 1],
       [0, 0, 0, 0, 1, 1, 1, 1],
       [0, 0, 0, 0, 1, 1, 1, 1],
       [0, 0, 0, 0, 1, 1, 1, 1],
       [0, 0, 0, 0, 1, 1, 1, 1],
       [2, 2, 2, 2, 3, 3, 3, 3],
       [2, 2, 2, 2, 3, 3, 3, 3]])
```
https://github.com/numpy/numpy/pull/11971

Reply:
also ,the example i have shows that it will even support non integer upscaling since the last "rows /columns" can be of different size, so long as they create this block-wise array.

Reply:
Nice, I did not know about `np.block` yet!

Reply:
FYI `np.block` concatenates over the last dimensions, so I had to nest the tiles in a list to make it work as intended.